---
uint32 dim
BlockParams[] poses
